src/briefcase/integrations/android_sdk.py

```diff
@@ -344,20 +340,27 @@ def upgrade(self):
 """
             ) from e
 
-    def list_packages(self):
-        """List the packages currently manged by the Android SDK."""
+    def list_packages(self, show_to_user=True):
+        """List the packages currently managed by the Android SDK.
+
+        :param show_to_user: True (default) to print installed tools to terminal.
+            False will only print installed tools to debug log.
+        """
         try:
-            # Using subprocess.run() with no I/O redirection so the user sees
-            # the full output and can send input.
-            self.command.subprocess.run(
-                [os.fsdecode(self.sdkmanager_path), "--list_installed"],
-                env=self.env,
-                check=True,
+            output = self.command.subprocess.check_output(
+                [os.fsdecode(self.sdkmanager_path), "--list_installed"], env=self.env
             )
         except subprocess.CalledProcessError as e:
             raise BriefcaseCommandError(
                 "Unable to invoke the Android SDK manager"
             ) from e
+        if show_to_user:
```

Comment:
Is this complication needed? AFAICT, `show_to_user` is always an effective proxy of "debug enabled". If debug is enabled, the output of the command will be in the log by virtue of being in debug mode; if debug isn't disabled, show_to_user will be false. AFACT, this has the effect of outputting the list of packages twice in debug mode, and doing nothing in non-debug mode.

Reply:
Yeah....I went round and round with what to do with this function because it has competing priorities between potentially showing the user the list of installed tools (in some potential future) and ensuring those tools are always listed in the debug output.

Since I probably shouldn't be writing branches for potential futures....I just updated the function to always only write the list of installed tools to debug output.